### ISSUE TITLE
Custom Infantry Armor Types and Divisors do not work properly. 

### ISSUE THREAD

**Issue by BrandonSmith16** [MegaMek/megamek#6090]
### Prerequisites and Pre-Issue Checklist

- [X] I'm reporting the issue to the correct repository:
- [MegaMek](https://github.com/MegaMek/megamek/issues)
- [MegaMekLab](https://github.com/MegaMek/megameklab/issues)
- [MekHQ](https://github.com/MegaMek/mekhq/issues)

- [X] I've tested the issue against at least the latest MILESTONE version
- [X] I've asked on the MegaMek Discord about the error
- [X] I've reviewed the BattleTech rules and MegaMek documentation, and I've confirmed that something isn't working as intended.
- [X] I've searched the Github tracker and haven't found the issue listed

### Severity *

Medium (Gameplay Limitation): Non-core functionality is impaired, providing a suboptimal but playable experience.

### Brief Description *

As of now custom armor divisors don't register at all in megamek or mekhq and custom armor types only sort of register properly in megmeklab.

You can only use them in megameklab if you use the following workaround: 
First you remove the armor using the remove armor button. Then you select an existing "donor" armor type in the stats sheet, then select use custom. _It won't allow you to apply the custom armor without selecting a base armor in the stats menu._ 
Trying to add special armor abilities like camo now won't work. To get the custom armor types to register you have to first click the "add armor" button _then_ select the special types. The custom divisor won't work at all it will only ever use the divisor of the base armor you selected in the stats menu. The partially custom armor can be saved to the .blk and will load in megameklab just fine after this, even if the divisor doesn't work. 

However it won't import the custom armor into megamek, just the donor armor. MekHQ will display the purchased custom unit correctly until you assign soldiers to it, then it will behave the same as megamek, only displaying the divisor and abilities of the base armor. 

Finally, once in megamek, attempting to select a custom armor type will not use the divisor you type in instead defaulting to some value  whenever you select a special armor ability. The abilities selected will however persist. 

### Steps to Reproduce

In order to find the bug in mekhq you have to purchase a unit with custom armor created via the workaround and then assign a unit. it will import properly but it will reset the armor to the base type and stats of the base armor once you assign soldiers. 

in order to reproduce the bug in megamek you merely need to load an infantry unit of any kind and attempt to assign a custom armor. 

### Operating System *

Windows 10

### Java Version *

17.7

### MegaMek Suite Version *

v0.50.0

### Final Checklist

- [X] I've checked to make sure that this issue has not already been filed
- [X] I'm reporting only one issue in this ticket for clarity and focus

**Comment by HammerGS** [comment]
What do you mean by "custom armor types"  Do you have screenshot.

**Comment by BrandonSmith16** [comment]
I mean if you were to try to create a suit with say an ecm lining and an
armor divisor of 2. DEST/Space/ECM/IR etc etc.

**Comment by gsparks3** [comment]
> What do you mean by "custom armor types" Do you have screenshot.

For reference - in 0.50 MegaMekLab, custom armor kits can be defined:
![image](https://github.com/user-attachments/assets/8522ae5c-6268-4163-adc3-c8c39c019a25)

You can also Configure a custom armor kit in the MegaMek lobby:
![image](https://github.com/user-attachments/assets/005b5baf-4604-4586-823b-100c72221165)

@BrandonSmith16 Was this specifically the "Configure in lobby" option having issues, or a pre-generated unit with a custom armor kit, or both?

**Comment by BrandonSmith16** [comment]
It was both.

**Comment by SJuliez** [comment]
Are custom armors allowed by the rules?

**Comment by pavelbraginskiy** [comment]
> Are custom armors allowed by the rules?

I don't think so, but I ended up fixing them a while back anyway and I believe they do work currently.
